packages/config/src/mutations/update.js

```diff
@@ -68,6 +68,8 @@ const deleteRedirectsFile = async function (redirectsPath, normalizedInlineConfi
 const backupConfig = async function ({ buildDir, configPath, redirectsPath }) {
   const tempDir = getTempDir(buildDir)
   await makeDir(tempDir)
+  // this makes sure we don't restore stale files
+  await Promise.all([deleteNoError(`${tempDir}/netlify.toml`), deleteNoError(`${tempDir}/_redirects`)])
```

Comment:
Good catch.
Should this be moved as `deleteNoError(dest)` at the beginning of `copyIfExists()`? This would avoid repeating the file paths.

Reply:
Yes, good point, I'll move those.

Reply:
Done in https://github.com/netlify/build/pull/3379/commits/ca1c9a4a82091e1e2b7c5664eae38ab66e559cc8.
I also renamed the function to `backupFile`